chain-vote/src/gang/babystep.rs

```diff
@@ -0,0 +1,107 @@
+#[cfg(not(feature = "zerocaf"))]
+use super::p256k1::*;
+#[cfg(feature = "zerocaf")]
+use super::zerocaf::*;
+use lazy_static::*;
+use rayon::prelude::*;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+// make steps asymmetric, in order to better use caching of baby steps.
+// balance of 2 means that baby steps are 2 time more than sqrt(MAX_VOTES_BSGS)
+const BALANCE: u64 = 2;
+// change this to adjust precomputed range, given this number it is possible to
+// find mupltiples up to SQRT_STEP_SIZE ^ 2 - 1
+lazy_static! {
+    static ref SQRT_STEP_SIZE: u64 = {
+        let max_votes = option_env!("MAX_VOTES_BSGS")
+            .unwrap_or("31999999999") // 1e10 -1
+            .parse::<u64>()
+            .expect("Invalid MAX_VOTES_BSGS: expected a integer");
+        ((max_votes as f64).sqrt() + 1.0) as u64
+    };
```

Comment:
This is a lot of hidden parameterization for a library.

I wonder if we could make this table an explicit object in the public API of chain-vote, that you generate at a place and with parameters of your choosing and pass to the `result` function (which should be renamed `tally_result` to emphasize that it does a lot of work and is not just some trivial data getter)?

Reply:
> with parameters of your choosing

Something like `PrivateTallyTable::new(total_voting_power)` which would get the equivalent of your `MAX_VOTES_BSGS` environment variable.

Reply:
Yes, maybe a builder for it (builder pattern or a function). Making the magic numbers some documented constants in the babystep API.

Reply:
Agreed, it was simpler when I started building it but then I noticed we need more options and it definitely grew out of its place.